portal-impl/src/com/liferay/portal/servlet/filters/virtualhost/VirtualHostFilter.java

```diff
@@ -312,18 +312,16 @@ protected void processFilter(
 					return;
 				}
 
+				String homeURL = PortalUtil.getRelativeHomeURL(
+					httpServletRequest);
+
 				if (Objects.equals(
 						group.getGroupKey(),
 						PropsValues.VIRTUAL_HOSTS_DEFAULT_SITE_NAME) &&
 					friendlyURL.equals(StringPool.SLASH) &&
-					!layoutSet.isPrivateLayout()) {
-
-					String homeURL = PortalUtil.getRelativeHomeURL(
-						httpServletRequest);
+					!layoutSet.isPrivateLayout() && _isApplyHomeURL(homeURL)) {
 
-					if (Validator.isNotNull(homeURL)) {
-						friendlyURL = homeURL;
-					}
+					friendlyURL = homeURL;
```

Comment:
My suggestion for this case is keep the inside homeURL and use the condition with
```
if (Validator.isNotNull(homeURL)&&!homeURL.equals(StringPool.SLASH){
...
}
```